### PR TITLE
fix(cli/rt): handle URL paths in mkdir

### LINKED ISSUE #8036: Error when using Deno.mkdir on Windows with absolute path
## Version

- deno 1.4.6
- v8 8.7.220.3
- typescript 4.0.3

## Problem

```
const dirname = new URL('../../data', import.meta.url);
console.log(dirname.toString());
await Deno.mkdir(dirname, {recursive: true});
```

image

Relative paths seem to work, but not absolute paths.

### PR DESCRIPTION
**Pull request by ross-weir**
fixes #8036 

Ensure paths created by `URL('../path')` are handled in `mkdir`.

### CODE REVIEW

File: cli/tests/unit/mkdir_test.ts [hunk 1]
```diff
@@ -197,3 +197,31 @@ unitTest(
     }
   },
 );
+
+unitTest(
+  { perms: { read: true, write: true } },
+  function mkdirSyncRelativeUrlPath(): void {
+    const testDir = Deno.makeTempDirSync();
+    const nestedDir = testDir + "/nested";
+    const path = new URL(nestedDir + "../dir");
```
Comment (coyotte508) [inline cli/tests/unit/mkdir_test.ts:206]: Should try `const path = new URL("../dir", nestedDir);`

File: cli/tests/unit/mkdir_test.ts [hunk 2]
```diff
@@ -197,3 +197,31 @@ unitTest(
...
+    assertDirectory(testDir + "/dir");
+  },
+);
+
+unitTest(
+  { perms: { read: true, write: true } },
+  async function mkdirRelativeUrlPath(): Promise<void> {
+    const testDir = Deno.makeTempDirSync();
+    const nestedDir = testDir + "/nested";
+    const path = new URL(nestedDir + "../dir");
```
Comment (coyotte508) [inline cli/tests/unit/mkdir_test.ts:220]: Same comment

**Review by coyotte508 — CHANGES_REQUESTED** [review]

File: cli/tests/unit/mkdir_test.ts (hunk 1 above)
Comment (ross-weir) [reply cli/tests/unit/mkdir_test.ts]: Thanks!

**Comment by ry** [comment]
I confirmed that the tests fail on master.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @ross-weir

**Pull request merged**